pkg/trace/writer/trace.go

```diff
@@ -80,6 +82,9 @@ func NewTraceWriter(cfg *config.AgentConfig, in <-chan *SampledSpans) *TraceWrit
 		// default to 50% of maximum memory.
 		qsize = int(math.Max(1, cfg.MaxMemory/2/float64(maxPayloadSize)))
 	}
+	if s := cfg.TraceWriter.FlushPeriodSeconds; s != 0 {
+		tw.tick = time.Duration(s*1000) * time.Millisecond
```

Comment:
This is to allow fractions of a second (e.g. in integrations we use `0.1`)

Reply:
I'd argue the following is equivalent but clearer?

```suggestion
		tw.tick = time.Duration(s*float64(time.Second))
```